include/RevOpts.h

```diff
@@ -91,18 +124,29 @@ class RevOpts {
   }
 
 private:
-  uint32_t numCores{};   ///< RevOpts: number of initialized cores
-  uint32_t numHarts{};   ///< RevOpts: number of harts per core
-  int      verbosity{};  ///< RevOpts: verbosity level
-
-  std::unordered_map<uint32_t, uint64_t>     startAddr{};      ///< RevOpts: map of core id to starting address
-  std::unordered_map<uint32_t, std::string>  startSym{};       ///< RevOpts: map of core id to starting symbol
-  std::unordered_map<uint32_t, std::string>  machine{};        ///< RevOpts: map of core id to machine model
-  std::unordered_map<uint32_t, std::string>  table{};          ///< RevOpts: map of core id to inst table
-  std::unordered_map<uint32_t, uint32_t>     prefetchDepth{};  ///< RevOpts: map of core id to prefretch depth
-  std::vector<std::pair<uint32_t, uint32_t>> memCosts{};       ///< RevOpts: vector of memory cost ranges
-  std::vector<std::string>                   Argv{};           ///< RevOpts: vector of function arguments
-  std::vector<std::string>                   MemDumpRanges{};  ///< RevOpts: vector of function arguments
+  uint32_t const numCores;   ///< RevOpts: number of initialized cores
+  uint32_t const numHarts;   ///< RevOpts: number of harts per core
+  int const      verbosity;  ///< RevOpts: verbosity level
+
+  // init all the standard options
+  // -- startAddr = 0x00000000
+  // -- machine = "G" aka, "IMAFD"
+  // -- table = internal
+  // -- memCosts[core] = 0:10
+  // -- prefetch depth = 16
+  // -- pipeLine = 5 ???
+
+  // clang-format off
+  std::vector<uint64_t>                     startAddr{decltype( startAddr     )( numCores, 0 )};                 ///< RevOpts: starting address
+  std::vector<std::string>                    machine{decltype( machine       )( numCores, "G" )};               ///< RevOpts: machine model
+  std::vector<std::string>                      table{decltype( table         )( numCores, "_REV_INTERNAL_" )};  ///< RevOpts: inst table
+  std::vector<std::pair<uint32_t, uint32_t>> memCosts{decltype( memCosts      )( numCores, {0, 10} )};           ///< RevOpts: memory cost range
+  std::vector<uint32_t>                 prefetchDepth{decltype( prefetchDepth )( numCores, 16 )};                ///< RevOpts: prefretch depth
+  // clang-format on
```

Comment:
All of the vectors which are initialized to default values are initialized at constructor time, after `numCores` declared above them is initialized. The `decltype()` syntax is syntactic sugar for not having to repeat the type in the far-left column a second time, which we need in order to make direct constructor calls to `std::vector` rather than initialization-list calls to `std::vector`, which have a different meaning.